presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -233,52 +239,78 @@ public void testIOExplain()
                                                         ImmutableSet.of(
                                                                 new FormattedRange(
                                                                         new FormattedMarker(Optional.of("1"), EXACTLY),
-                                                                        new FormattedMarker(Optional.of("199"), EXACTLY)))))))),
-                        Optional.of(new CatalogSchemaTableName(catalog, "tpch", "test_orders"))));
+                                                                        new FormattedMarker(Optional.of("199"), EXACTLY)))))),
+                                estimate)),
+                        Optional.of(new CatalogSchemaTableName(catalog, "tpch", "test_orders")),
+                        estimate));
 
         assertUpdate("DROP TABLE test_orders");
     }
 
+    private static class TypeAndEstimate
+    {
+        public final Type type;
+        public final EstimatedStatsAndCost estimate;
+
+        public TypeAndEstimate(Type type, EstimatedStatsAndCost estimate)
+        {
+            this.type = type;
+            this.estimate = estimate;
+        }
+    }
+
     @Test
     public void testIoExplainWithPrimitiveTypes()
     {
-        Map<Object, Type> data = new HashMap<>();
-        data.put("foo", VarcharType.createUnboundedVarcharType());
-        data.put(Byte.toString((byte) (Byte.MAX_VALUE / 2)), TinyintType.TINYINT);
-        data.put(Short.toString((short) (Short.MAX_VALUE / 2)), SmallintType.SMALLINT);
-        data.put(Integer.toString(Integer.MAX_VALUE / 2), IntegerType.INTEGER);
-        data.put(Long.toString(Long.MAX_VALUE / 2), BigintType.BIGINT);
-        data.put(Boolean.TRUE.toString(), BooleanType.BOOLEAN);
-        data.put("bar", CharType.createCharType(3));
-        data.put("1.2345678901234578E14", DoubleType.DOUBLE);
-        data.put("123456789012345678901234.567", DecimalType.createDecimalType(30, 3));
-        data.put("2019-01-01", DateType.DATE);
-        data.put("2019-01-01 23:22:21.123", TimestampType.TIMESTAMP);
-        for (Map.Entry<Object, Type> entry : data.entrySet()) {
-            @Language("SQL") String query = format(
+        EstimatedStatsAndCost estimate10 = new EstimatedStatsAndCost(1.0, 10.0, 10.0, 0.0, 0.0, 3.0);
+        EstimatedStatsAndCost estimate11 = new EstimatedStatsAndCost(1.0, 11.0, 11.0, 0.0, 0.0, 3.0);
+        EstimatedStatsAndCost estimate13 = new EstimatedStatsAndCost(1.0, 13.0, 13.0, 0.0, 0.0, 3.0);
+        EstimatedStatsAndCost estimate16 = new EstimatedStatsAndCost(1.0, 16.0, 16.0, 0.0, 0.0, 6.0);
+        EstimatedStatsAndCost estimate17 = new EstimatedStatsAndCost(1.0, 17.0, 17.0, 0.0, 0.0, 3.0);
+        EstimatedStatsAndCost estimate25 = new EstimatedStatsAndCost(1.0, 25.0, 25.0, 0.0, 0.0, 3.0);
+        Map<Object, TypeAndEstimate> data = new LinkedHashMap<>();
```

Comment:
why linked hash map?

Reply:
It's a linked hash map because the assert below prints the index of the entry upon failure to make the entry easier to find.

Reply:
We usually use `ImmutableMap.Builder` also when we want to preserve insertion order.
ImmutableMap guarantees order.

BTW here `List<Entry<..>` would do. Or restructuring the test so that `data` is not needed at all.